products/accredited-representation-management/technical-documentation/appoint-a-representative/poa_request_proposal.md

```diff
@@ -0,0 +1,160 @@
+
+## Problem
+In order to create a functional user interface for accredited representatives ([see the sample UI](#sample-ui)), the POA Request objects must support reasonably efficient searching or filtering on the following Veteran attributes: first name, last name, city, state, zip code, and consent limits. The front end must also be able to display information related to change of address authorization and health record access limitations.
+
+There is not an existing pattern or solution in vets-api for storing Veteran PII that is secure and efficiently searchable for a many row result.
+
+## Solution
+Code below is for illustrative purposes. These snippets are not likely to be the final implementation.
+
+### Column level encryption
+The [blind index gem](https://github.com/ankane/blind_index) allows for exact matching queries on encrypted columns. This will be used on fields that users will search or filter by. Existing KMS encryption will be used on the fields that are for display purposes only.
+
+#### The Model
+```ruby
+class PowerOfAttorneyRequest < ApplicationRecord
+  has_kms_key
+  has_encrypted :consent_data, key: :kms_key, **lockbox_options
+  blind_index :first_name, :last_name, :city, :state_code, :zip_code
+
+  belongs_to :user_account
+  belongs_to :requested_individual, class_name: 'AccreditedIndividual'
+  belongs_to :accepted_or_declined_by, class_name: 'AccreditedIndividual', optional: true
+
+  enum status: {
+    'pending' => 'pending',
+    'accepted' => 'accepted',
+    'declined' => 'declined',
+    'expired' => 'expired'
+  }
+
+  def accept!(portal_user)
+    terminate
+    self.status = 'accepted'
+    self.accepted_or_declined_by = portal_user
+
+    self.save
+  end
+
+  def decline!(portal_user, reason)
+    terminate
+    self.status = 'declined'
+    self.accepted_or_declined_by = portal_user
+    self.decline_reason = reason
+
+    self.save
+  end
+
+  def expire!
+    terminate
+    self.status = 'expired'
+
+    self.save
+  end
+
+  private
+
+  def terminate
+    self.first_name = nil
+    self.last_name = nil
+    self.city = nil
+    self.state_code = nil
+    self.zip_code = nil
+    self.consent_data = nil
+  end
+end
+```
+
+#### The Migration
+```ruby
+class CreatePowerOfAttorneyRequests < ActiveRecord::Migration[7.1]
+  def change
+    create_table :power_of_attorney_requests do |t|
+      t.references :user_account, foreign_key: true, type: :uuid
+      t.references :requested_individual, foreign_key: {to_table: 'accredited_individual'}, type: :uuid
+
+      t.string :first_name_bidx
+      t.string :last_name_bidx
+      t.string :city_bidx
+      t.string :state_bidx
+      t.string :zip_code_bidx
+
+      t.text :consent_data_ciphertext
+
+      t.string :status
+      t.datetime :expires_at
```

Comment:
I'm wondering if we need `expired_at`?

Might it be more flexible to set a constant for the number of days a request expires after `created_at`?